UserInterface/fileOutput/outputFile_class.f90

```diff
@@ -210,17 +211,54 @@ module outputFile_class
   !!
   !! Args:
   !!   type [in] -> Character to specify the format of the output file
-  !!   fatalErrors [in] -> Optional. Set on/off fatalErrors oninvalid use of output file.
+  !!   fatalErrors [in] -> Optional. Set on/off fatalErrors on invalid use of output file.
   !!                         Default is ON.
-  !!
-  subroutine init(self, type, fatalErrors)
-    class(outputFile), intent(inout)     :: self
-    character(*),intent(in)              :: type
-    logical(defBool),optional,intent(in) :: fatalErrors
+  !!   filename [in] -> Optional. Set the name of the output file. If none is given no output is
+  !!                    written. Must be given without extension. It will be added automatically
+  !!                    based on the format of the output file.
+  !!
+  subroutine init(self, type, fatalErrors, filename)
+    class(outputFile), intent(inout)      :: self
+    character(*), intent(in)              :: type
+    logical(defBool), optional,intent(in) :: fatalErrors
+    character(*), optional, intent(in)    :: filename
+    integer(shortInt)                     :: error
+    character(99)                         :: errorMsg
+    logical(defBool)                      :: isOpen
+    character(100), parameter             :: Here = "init (outputFile_class.f90)"
 
     self % type = type
     allocate( self % output, source = new_asciiOutput(self % type))
 
+    ! Open file if given
+    ! Otherwise make sure that the unit is not opened
+    if (present(filename)) then
+      ! Because GFotran hates implicit allocation
+      ! It causes buggy warnings to appear in compilation
+      allocate(self % outputFileName, source = trim(filename) // "." // self % output % extension())
+
+      open(newunit = self % outputUnit, &
```

Comment:
It looks like 'error' is gone from this list.

Reply:
same with errorMsg

Reply:
:fearful: Ups... not great thing to forget

Reply:
Should be fixed.